java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -199,8 +200,6 @@ public static class TokenRefreshHandlerImpl implements OAuth2CredentialsWithRefr
         private static final int MAX_TOKEN_REFRESH_ATTEMPTS = 3;
         private static final long WAIT_AFTER_FAILED_LOCK_ATTEMPTS = 2000L;
 
-        private AccessToken currentToken = null;
```

Comment:
new instance of `TokenRefreshHandlerImpl` created for each request; so what's the point of storing `currentToken` here? 

risk is that in 401 case we still have the `currentToken` that gave the 401, so we'll enter a loop with it.